float64 gap_direction